src/list/reducers.js

```diff
@@ -37,7 +37,13 @@ export function cacheReducer(state = {items: [], currentItem: null}, action) {
   case actions.ADD_ITEM_COMPLETED:
     return {
       ...state,
-      items: [...state.items, makeItemSummary(action.item)],
+      items: [
+        // HACK: add dispatched via Logins API events may be a duplicate
```

Comment:
Small hacky wrinkle here: Since I'm broadcasting datastore changes based on Logins API events without any source, the UI source that made the changes will get its own change dispatched back to it. This will look like a duplicate. So, I papered over that by making the "add" action idempotent here.